src/jiraCommon.gs

```diff
@@ -651,13 +649,121 @@ function getIssue(issueKey, fields) {
   };
 
   var request = new Request();
-  request.call('issueStatus', {issueIdOrKey: issueKey, fields: fields})
+  var requestParams = {issueIdOrKey: issueKey};
+  if (fields!= null) {
+    requestParams["fields"] = fields;
+  }
+  request.call('issueStatus', requestParams)
     .withSuccessHandler(ok)
     .withFailureHandler(error);
   
   return response;
 }
 
+/**
+ * Takes the native JSON response from JIRA for a field definition and returns an object
+ * with the fields used in the Application 
+ * @param jiraFieldResponse - the response from JIRA
+ * @return array Array of objects for each field 
+return {
+          key:        cField.key || cField.id, // Server API returns ".id" only while Cloud returns both with same value
+          name:       cField.name,
+          custom:     cField.custom,
+          schemaType: _type,
+          supported:  (arrSupportedTypes.indexOf(_type) > -1)
+        };
+ */
+function convertJiraFieldResponseToFieldRecord(jiraFieldResponse) {
+  var arrSupportedTypes = ['string', 'number', 'datetime', 'date', 'option', 'array|option', 'array|string', 'user', 'array|user', 'group', 'array|group', 'version', 'array|version'];
```

Comment:
If you can explain the logic of whats happening in those lines I will refactor it all to use getAllJiraFields

Reply:
I think it was because, EPIC consists of more than one field object, it has a label and link, which i must retrieve from 2 different fields. This also why i have a custom EPIC object here:
https://github.com/ljay79/jira-tools/blob/develop/src/customFields.gs#L6

to store the reference for back and forth...

Reply:
OK got it - Refactor is here as an issue as it will involve more work than I expected. Once this PR Is closed I am happy to do it.
https://github.com/ljay79/jira-tools/issues/140